lib/Onyx.js

```diff
@@ -635,6 +636,24 @@ function hasPendingMergeForKey(key) {
     return Boolean(mergeQueue[key]);
 }
 
+/**
+ * When merging 2 objects into onyx that contain an array, we want to completely replace the array instead of the default
+ * behavior which is to merge each item by its index.
+ * ie:
+ * merge({a: [1]}, {a: [2,3]}):
+ *  - In the default implementation would produce {a:[1,3]}
+ *  - With this function would produce: {a: [2,3]}
+ * @param {*} objValue
+ * @param {*} srcValue
+ * @return {*}
+ */
+// eslint-disable-next-line rulesdir/prefer-early-return
+function customizerForMergeWith(objValue, srcValue) {
+    if (_.isArray(objValue)) {
+        return srcValue;
+    }
+}
```

Comment:
NAB, I'm new to `mergeWith()`. Any idea if this customizer work only on top level array values or is it deep? What would be the expected behavior for something like this:

```js
const existing = {a: {arr: [1, 2, 3]}};
const new = {a: {arr: [4, 5, 6]}};
```

Reply:
It is used on all levels, so the example above works as you expect (final result would be `{a: {arr: [4,5,6]}}`)

Reply:
Nice. Much more predictable.